src/Avalonia.Controls/ApplicationLifetimes/ClassicDesktopStyleApplicationLifetime.cs

```diff
@@ -123,6 +129,14 @@ public void Dispose()
             if (_activeLifetime == this)
                 _activeLifetime = null;
         }
+        
+        private void ShutdownRequested(object sender, CancelEventArgs e)
+        {
+            foreach (var w in Windows)
+                w.Close();
```

Comment:
hey guys! it's kinda breaking the logic that was added in this PR a while ago - https://github.com/AvaloniaUI/Avalonia/pull/5616 
With that logic we are able to cancel closing event from parent level. But current PR do it in a different way - just iterates throught every window and goes straight to the closing and we have no chance to cancel it from parent level

Reply:
Yeah it seems that #5616 added the logic to `HandleClosing` which means it only gets run when the user clicks the close button, not when closed programmatically. Was this intended @danwalmsley ?

Reply:
at that point HandleClosing triggered for CMD + Q too, not only close button. But this new PR changes the behavior of CMD + Q and it's now the same as quit from dock

Reply:
Yeah the real issue is that the logic isn't invoked when calling `Window.Close`

Reply:
We should still only call close on top level windows right? Because all child windows will be closed by their parent.

Reply:
https://github.com/AvaloniaUI/Avalonia/pull/5616 added logic for parent windows to call close on child windows. This logic though was added to `HandleClosing` which reacts to a close notification coming from the OS, which means that calling Close from code bypasses this logic. Surely calling Close from code should query the child windows in the same way as the OS signaling close?

Reply:
Actually looks like I was wrong, `Close` should call `Closing` on child windows... Need to investigate more.

Reply:
I don't think it breaks the logic from #5616 to be honest, but now a shutdown will try to close _all_ windows in the application, even child windows. I think what you're asking for is that a shutdown only tries to close non-child windows, and that the parent window will be responsible for closing them if it doesn't cancel shutdown.

Reply:
Yep, this seems like correct behavior: cross checked with WPF and owned windows are closed by their owners on shutdown. Implemented.